src/nvim/msgpack_rpc/status_event.c

```diff
@@ -0,0 +1,74 @@
+#include <stdbool.h>
+#include <inttypes.h>
+
+#include "nvim/api/private/helpers.h"
+#include "nvim/map.h"
+#include "nvim/msgpack_rpc/channel.h"
+#include "nvim/msgpack_rpc/status_event.h"
+#include "nvim/memory.h"
+
+static PMap(uint64_t) *channels = NULL;
+
+/// Initializes the module
+void status_event_init(void)
+{
+  channels = pmap_new(uint64_t)();
+}
+
+void status_event_subscribe(uint64_t channel_id) {
+  StatusInfo **ref = (StatusInfo **)pmap_ref(uint64_t)(channels, channel_id, true);
+  if (!*ref) {
+    *ref = xmalloc(sizeof(**ref));
+  }
+  StatusInfo *status = *ref;
+  status->last_curbuf = -1;
+  status->last_curwin = -1;
+  status->last_curtab = -1;
+}
+
+void status_event_unsubscribe(uint64_t channel_id) {
+  StatusInfo *status = pmap_del(uint64_t)(channels, channel_id);
+  if (status) {
+    xfree(status);
+  }
+}
+
+
+static void update(uint64_t channel_id, StatusInfo *status) {
```

Comment:
It should happen every time control is yeilded to the clients, no matter request, response or notification (or to all clients when nvim is ready for events from all).

It explicitly does not say "this and this happened because of request" only "these core properties of the editor is different since the last update"